src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -653,6 +653,23 @@ void
 PeerImp::doProtocolStart()
 {
     onReadMessage(error_code(), 0);
+
+    protocol::TMManifests tm;
+    tm.set_history (true);
+
+    overlay_.manifestCache ().for_each_manifest (
```

Comment:
@nbougalis I addressed this concern by reserving memory for the list. This should make the current code similar in performance to copying from a cache.